src/theme/use-theme-mode.ts

```diff
@@ -0,0 +1,51 @@
+import { useContext } from "react";
+import { ThemeContext } from "styled-components";
+import { ColourMode, ThemeSpec } from "./types";
+
+export interface DSTheme {
+    colourMode: ColourMode;
+    theme: ThemeSpec | null;
+}
+
+// Hook to access the current theme mode and theme object
+export const useDSTheme = (): DSTheme => {
+    const theme = useContext(ThemeContext) as ThemeSpec;
+
+    if (!theme) {
+        console.warn(
+            "useDSTheme must be used within a DSThemeProvider or ThemeProvider"
+        );
+        return {
+            colourMode: "light" as ColourMode,
```

Comment:
casting not needed as Typescript can pick up the type from the functions' return value

```suggestion
            colourMode: "light",
```